atc/engine/builder/delegate_factory.go

```diff
@@ -389,6 +390,20 @@ func (delegate *buildStepDelegate) buildOutputFilter(str string) string {
 	return it.line
 }
 
+func (delegate *buildStepDelegate) ImageSourceRedaction(source atc.Source) (atc.Source, error) {
```

Comment:
This is a clever approach to the problem! There are a couple caveats though: it relies on the credentials coming from a credential manager (or `load_var`), and it relies on having secret redaction enabled.

I think these caveats are OK, but they are probably worth reflecting in the RFC.

We should also look into whether we can un-feature-flag secret redaction. I've opened an issue for that: https://github.com/concourse/concourse/issues/5771

All that aside, I think this might be a clearer name:

```suggestion
func (delegate *buildStepDelegate) RedactImageResource(source atc.Source) (atc.Source, error) {
```

Reply:
FWIW, I'm willing to concede that what you had before was simpler and safer, even though it opened the "black box".

If this path starts to feel too treacherous I'd be OK with going back and opening an issue for something like https://github.com/concourse/concourse/pull/5034#discussion_r436028270 to clean it up.

Reply:
Renamed.